web/js/containers/sidebar/smart-handoff.js

```diff
@@ -0,0 +1,470 @@
+
+    if (selectedLayer.daynight) urlTotalGranules += `&day_night_flag=${selectedLayer.daynight}`;
+
+    let urlSelectedGranules = urlTotalGranules;
+
+    // Gets the total amount of granules that the layer has
+    totalGranules = await fetch(urlTotalGranules, { timeout: 5000 })
+      .then(async(response) => {
+        const result = await response.json();
+        return result.feed.facets.children[0].children[0].children[0].count;
+      })
+      .catch((error) => 0);
+
+    // Gets the total subset of granules that are within the defining bounding box
+    if (showBoundingBox) {
+      urlSelectedGranules += `&bounding_box=${currentExtent.southWest},${currentExtent.northEast}`;
+      selectedGranules = await fetch(urlSelectedGranules, { timeout: 5000 })
+        .then(async(response) => {
+          const result = await response.json();
+          return result.feed.facets.children[0].children[0].children[0].count;
+        })
+        .catch((error) => 0);
+    }
+
+    this.setState({ selectedGranules, totalGranules });
+    this.setState({ isSearchingForGranules: false });
+  }
+
+  /**
+   * Default render which displays the data-download panel
+   */
+  render() {
+    const {
+      screenWidth,
+      screenHeight,
+      activeLayers,
+      isActive,
+      showWarningModal,
+      selectedDate,
+    } = this.props;
+
+    const { selectedLayer, isSearchingForGranules } = this.state;
+    const {
+      selectedGranules, totalGranules, coordinates, currentExtent, showBoundingBox,
+    } = this.state;
+
+    // Determine if data-download 'smart-handoff' tab is activated by user
+    if (!isActive) return null;
+
+    const { boundaries } = this.state;
+    const {
+      x, y, x2, y2,
+    } = boundaries;
+
+    // Used to determine if the added smart-handoff modal should be shown
+    const { HIDE_EDS_WARNING } = safeLocalStorage.keys;
+    let showModal = safeLocalStorage.getItem(HIDE_EDS_WARNING);
+    showModal = true;
+
+    // Determine if the download button is enabled
+    const isValidDownload = selectedLayer.id !== undefined;
+
+    const availableLayers = activeLayers.filter((layer) => layer.conceptId !== undefined).length;
+    const areThereLayersToDownload = availableLayers > 0;
+
+    if (areThereLayersToDownload) {
+      return (
+        <div id="smart-handoff-side-panel">
+
+          <h1>Select an available layer to download:</h1>
+
+          <div id="esd-notification">
+            Downloading data will be performed using
+            {' '}
+            <a href="https://search.earthdata.nasa.gov" target="_blank" rel="noopener noreferrer">NASA&apos;s Earthdata Search</a>
+            {' '}
+            application.
+          </div>
+
+          <hr />
+
+          <div id="smart-handoff-layer-list">
+            {activeLayers.map((layer, index) => {
+              if (layer.conceptId) {
+                return (
+                  <div className="layer-item">
+                    <input
```

Comment:
This `<input>` element needs to have a unique `key` attribute:
![Screen Shot 2020-11-23 at 8 25 08 AM](https://user-images.githubusercontent.com/1480000/99967266-9a621c80-2d65-11eb-8b8e-8c5c5c2e8aba.png)

Reply:
Ah yes, so originally I was leveraging the concept ID to each layer, but as I found out, concept IDs are not truly unique, multiple layer imagery can reference the same concept ID. I tried utilizing the index but the linter doesn't agree. Looks like I'll have to leverage the layer ID (i.e. AIRS_L2_Carbon_Monoxide_500hPa_Volume_Mixing_Ratio_Night).